src/fsharp/NameResolution.fs

```diff
@@ -2436,17 +2453,17 @@ let ResolveTypeLongIdent sink (ncenv:NameResolver) occurence fullyQualified nenv
 /// Resolve a long identifier representing a record field in a module or namespace
 let rec ResolveFieldInModuleOrNamespace (ncenv:NameResolver) nenv ad (resInfo:ResolutionInfo) depth m (modref: ModuleOrNamespaceRef) _mty (lid: Ident list) = 
     let typeNameResInfo = TypeNameResolutionInfo.Default
+    let returnError id = raze (UndefinedName(depth,FSComp.SR.undefinedNameRecordLabelOrNamespace,id,[],0,None))
```

Comment:
I think remove this particular inner function for now and just copy it into each of the error cases, it doesn't really add anything to make it a function.